### PR TITLE
fix(orchestrator): prefer live integrations over memory_tree for inbox/doc queries

### PR DESCRIPTION
**Pull request by senamakel**
## Summary

- Orchestrator was answering "any new emails?" / "summarise the notion doc I edited" from the retrospective `memory_tree` index instead of calling the live integration. Users had to explicitly say "search my gmail" to force the right behaviour.
- Reworked the orchestrator prompt to disambiguate `delegate_to_integrations_agent` (live SaaS) from `memory_tree` (historical ingest), with an explicit decision-tree step that catches inbox/doc/calendar/chat phrasings before memory tools are considered.
- No tool-surface changes — prompt-only fix, plus a test update for the new decision-tree wording.

## Problem

Users complained they had to spell out "search my gmail" or "check my inbox" to get the orchestrator to actually hit the Gmail integration. Otherwise it would happily answer from `memory_tree` — which is a stale, retrospective summary of already-ingested content, not a live view of the inbox.

Root cause in the prompt:

1. The "Delegation Decision Tree" listed `memory_*` under "direct tools" in Step 2, before the "specialised execution" step that mentions `delegate_to_integrations_agent`. Combined with the bias toward direct tools, the model preferred memory.
2. The "Memory tree retrieval" section described `memory_tree` as a tool to "query the user's ingested email/chat/document history" with a mode literally documented as "Use for 'in my email last week…' intents" — exactly the phrasing users were tripping on.
3. The Connected Integrations and Response Style example sections never gave a concrete live-integration delegation pattern to imitate.

## Solution

`src/openhuman/agent/agents/orchestrator/prompt.md`:

- Inserted a new **Step 2** in the decision tree that explicitly routes any request naming or implying a connected external service (email/inbox/gmail/calendar/notion/drive/slack/whatsapp/telegram/linear/etc.) to `delegate_to_integrations_agent` **before** memory or other direct tools are considered. Explicit guidance: "Do this even if `memory_tree` could plausibly answer."
- If the relevant toolkit is not in Connected Integrations, the orchestrator must point the user at Settings → Connections → [Service] rather than silently falling back to `memory_tree`.
- Reframed the **Memory tree retrieval** section as historical-only. It is now described as a retrospective index over already-ingested content, not a live API. Listed the phrasings that genuinely warrant it ("what did we discuss last month", "summarise my recent activity"). Removed misleading language about "in my email last week…".
- Added a concrete Gmail example to the Response Style section so the model has a delegation pattern to imitate, and annotated the existing Notion example as a live delegation.

`src/openhuman/agent/agents/orchestrator/prompt.rs`:

- Updated `build_includes_direct_first_decision_tree` to assert the new decision-tree wording (live-integration step + the "do this even if memory_tree could plausibly answer" guidance) so a regression that removed Step 2 would fail the build.

## Submission Checklist

- [x] Tests added or updated (happy path + at least one failure / edge case) per [Testing Strategy](../gitbooks/developing/testing-strategy.md#failure-path-requirement)
- [x] N/A: prompt-only change with no executable diff coverage signal; the existing `orchestrator::prompt::tests` suite was extended to assert the new decision-tree wording.
- [x] N/A: behaviour-only change — no new/removed/renamed feature rows in `docs/TEST-COVERAGE-MATRIX.md`.
- [x] N/A: no matrix feature IDs involved.
- [x] No new external network dependencies introduced (mock backend used per [Testing Strategy](../gitbooks/developing/testing-strategy.md#mock-policy))
- [x] N/A: no release-cut surfaces touched (orchestrator prompt only).
- [x] N/A: no linked issue — user-reported behaviour, fixed directly.

## Impact

- Runtime: desktop only (orchestrator agent prompt). No platform-specific code paths touched.
- Behaviour: users asking about inbox / docs / calendar / chat state in natural language ("any new emails from alice today?", "summarise the latest notion doc") will now get a live integration call instead of a stale memory summary.
- Compatibility: no schema or tool-surface change. `memory_tree` still available for retrospective queries.
- Risk: prompt regressions are caught by the updated unit tests in `prompt.rs`. Worst-case behaviour mismatch is recoverable via a follow-up prompt tweak — no migrations or persisted state involved.

## Related

- Closes:
- Follow-up PR(s)/TODOs:

---

### Linear Issue
- Key: N/A
- URL: N/A

### Commit & Branch
- Branch: fix/orchestrator-prefer-live-integrations
- Commit SHA: 74c2b0d44fb12439ed9dda4219e7e0c997e29249

### Validation Run
- [x] `pnpm --filter openhuman-app format:check` — N/A: no `app/` files touched.
- [x] `pnpm typecheck` — N/A: no TS files touched.
- [x] Focused tests: `cargo test --manifest-path Cargo.toml --lib orchestrator::prompt` — 7 passed, 0 failed.
- [x] Rust fmt/check (if changed): orchestrator prompt module compiles via the focused test run above.
- [x] Tauri fmt/check (if changed): N/A — no `app/src-tauri` changes.

### Validation Blocked
- command: N/A
- error: N/A
- impact: N/A

### Behavior Changes
- Intended behavior change: orchestrator routes live inbox/doc/calendar/chat requests to `delegate_to_integrations_agent` instead of `memory_tree`.
- User-visible effect: "any new emails?" hits Gmail; "what did we discuss last month" still hits memory.

### Parity Contract
- Legacy behavior preserved: `memory_tree` still callable for retrospective queries; all existing modes (`search_entities`, `query_topic`, `query_source`, `query_global`, `drill_down`, `fetch_leaves`) unchanged.
- Guard/fallback/dispatch parity checks: orchestrator tool registry (`agent.toml`) untouched; `agents::loader` invariant "orchestrator must have memory_tree" still holds.

### Duplicate / Superseded PR Handling
- Duplicate PR(s): N/A
- Canonical PR: N/A
- Resolution: N/A

## Summary by CodeRabbit

* **New Features**
  * Improved orchestrator agent routing to prioritize live external services and explicit integration delegation; examples and guidance updated.

* **Chores**
  * Bumped core dependency versions and adjusted a dependency feature for compatibility.
  * Mock API delay handling now enforces a configurable maximum cap to prevent excessive simulated delays.

[Review Change Stack](https://app.coderabbit.ai/change-stack/tinyhumansai/openhuman/pull/1731)

### CODE REVIEW

File: src/openhuman/agent/agents/orchestrator/prompt.md [hunk 1]
```diff
@@ -108,19 +112,33 @@ checking notion
...
+(`delegate_to_integrations_agent` with `toolkit: "notion"` — the user wants the live doc, not a memory summary.)
+
+User: any new emails from alice today?
+→
+```text
+checking gmail
+
+one, 2pm: "lunch friday?" — wants to grab food, no agenda
+```
+(`delegate_to_integrations_agent` with `toolkit: "gmail"`. Do **not** start with `memory_tree` — the user is asking about live inbox state.)
```
[the remark this answers is not in the document]
Comment (senamakel) [reply src/openhuman/agent/agents/orchestrator/prompt.md]: Fixed in 6a275759 — replaced the em-dashes in lines 122/124 with a comma and a semicolon. Other em-dashes in the file are pre-existing rule descriptions/headers (not example replies), so left alone.
